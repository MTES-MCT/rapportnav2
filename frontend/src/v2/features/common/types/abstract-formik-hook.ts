import { FormikErrors } from 'formik'
import { ObjectSchema } from 'yup'

export interface AbstractFormikHook<T, M> {
  initValue?: M
  errors?: FormikErrors<M>
  beforeInitValue: (value?: T) => M | undefined
  beforeSubmit: (value?: M, errors?: FormikErrors<M>) => T | undefined
  handleSubmit: (
    value?: M,
    errors?: FormikErrors<M>,
    onSubmit?: (valueToSubmit?: T) => Promise<unknown>
  ) => Promise<void>
}

export interface AbstractFormikSubFormHook<M> {
  initValue?: M
  errors?: FormikErrors<M>
  validationSchema?: ObjectSchema<any>
  handleSubmit: (value?: M, errors?: FormikErrors<M>) => Promise<void>
}
