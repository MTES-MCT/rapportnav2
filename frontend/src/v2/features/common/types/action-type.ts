export enum ActionType {
  CONTROL = 'CONTROL',
  NOTE = 'NOTE',
  SURVEILLANCE = 'SURVEILLANCE',
  STATUS = 'STATUS',
  CONTACT = 'CONTACT',
  RESCUE = 'RESCUE',
  OTHER = 'OTHER',
  NAUTICAL_EVENT = 'NAUTICAL_EVENT',
  VIGIMER = 'VIGIMER',
  ANTI_POLLUTION = 'ANTI_POLLUTION',
  BAAEM_PERMANENCE = 'BAAEM_PERMANENCE',
  PUBLIC_ORDER = 'PUBLIC_ORDER',
  REPRESENTATION = 'REPRESENTATION',
  ILLEGAL_IMMIGRATION = 'ILLEGAL_IMMIGRATION',
  INQUIRY = 'INQUIRY',
  COMMUNICATION = 'COMMUNICATION',
  TRAINING = 'TRAINING',
  UNIT_MANAGEMENT_PLANNING = 'UNIT_MANAGEMENT_PLANNING',
  UNIT_MANAGEMENT_TRAINING = 'UNIT_MANAGEMENT_TRAINING',
  UNIT_MANAGEMENT_OTHER = 'UNIT_MANAGEMENT_OTHER',
  CONTROL_SECTOR = 'CONTROL_SECTOR',
  CONTROL_NAUTICAL_LEISURE = 'CONTROL_NAUTICAL_LEISURE',
  CONTROL_SLEEPING_FISHING_GEAR = 'CONTROL_SLEEPING_FISHING_GEAR',
  OTHER_CONTROL = 'OTHER_CONTROL',
  RESOURCES_MAINTENANCE = 'RESOURCES_MAINTENANCE',
  MEETING = 'MEETING',
  PV_DRAFTING = 'PV_DRAFTING',
  HEARING_CONDUCT = 'HEARING_CONDUCT',
  LAND_SURVEILLANCE = 'LAND_SURVEILLANCE',
  FISHING_SURVEILLANCE = 'FISHING_SURVEILLANCE',
  MARITIME_SURVEILLANCE = 'MARITIME_SURVEILLANCE'
}

export enum ActionGroupType {
  OTHER_GROUP = 'OTHER_GROUP',
  CONTROL_GROUP = 'CONTROL_GROUP',
  SURVEILLANCE_GROUP = 'SURVEILLANCE_GROUP',
  ADMINISTRATIVE_GROUP = 'ADMINISTRATIVE_GROUP',
  UNIT_MANAGEMENT_GROUP = 'UNIT_MANAGEMENT_GROUP'
}
