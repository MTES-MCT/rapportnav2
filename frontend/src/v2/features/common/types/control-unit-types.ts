export type ControlUnitResource = {
  id: string
  controlUnit: ControlUnit
  controlUnitId: number
  isArchived: boolean
  name: string
  note?: string
  photo?: string
  station: Station
  stationId: number
  type: ControlUnitResourceType
}

export type Station = {
  id: number
  latitude: number
  longitude: number
  name: string
}

export type ControlUnit = {
  id: number
  areaNote?: string
  administrationId: number
  departmentAreaInseeCode?: string
  isArchived: boolean
  name: string
  termsNote?: string
}

export type Administration = {
  id: number
  name: string
  controlUnitIds: number[]
  isArchived: boolean
  controlUnits: ControlUnit[]
}

enum ControlUnitResourceType {
  AIRPLANE = "AIRPLANE",
  BARGE = "BARGE",
  CAR = "CAR",
  DRONE = "DRONE",
  EQUESTRIAN = "EQUESTRIAN",
  FAST_BOAT = "FAST_BOAT",
  FRIGATE = "FRIGATE",
  HELICOPTER = "HELICOPTER",
  HYDROGRAPHIC_SHIP = "HYDROGRAPHIC_SHIP",
  KAYAK = "KAYAK",
  LIGHT_FAST_BOAT = "LIGHT_FAST_BOAT",
  MINE_DIVER = "MINE_DIVER",
  MOTORCYCLE = "MOTORCYCLE",
  NET_LIFTER = "NET_LIFTER",
  NO_RESOURCE = "NO_RESOURCE",
  OTHER = "OTHER",
  PATROL_BOAT = "PATROL_BOAT",
  PEDESTRIAN = "PEDESTRIAN",
  PIROGUE = "PIROGUE",
  RIGID_HULL = "RIGID_HULL",
  SEA_SCOOTER = "SEA_SCOOTER",
  SEMI_RIGID = "SEMI_RIGID",
  SUPPORT_SHIP = "SUPPORT_SHIP",
  TRAINING_SHIP = "TRAINING_SHIP",
  TUGBOAT = "TUGBOAT"
}
