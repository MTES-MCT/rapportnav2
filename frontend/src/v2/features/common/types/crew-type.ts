import { Service } from '@common/types/crew-types.ts'

export type Agent = {
  id: number
  firstName: string
  lastName: string
  services: Service[]
}

export type AgentRole = {
  id: string
  title: string
}

export type MissionCrewMember = {
  id?: string
  agent: Agent
  comment?: string
  role?: AgentRole
  missionId?: number
}
