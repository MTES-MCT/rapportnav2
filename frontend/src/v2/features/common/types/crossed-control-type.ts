export enum CrossControlOriginType {
  OPPORTUNITY_CONTROL = 'OPPORTUNITY_CONTROL',
  FOLLOW_UP_CONTROL = 'FOLLOW_UP_CONTROL',
  CNSP_REPORTING = 'CNSP_REPORTING',
  OTHER_REPORTING = 'OTHER_REPORTING',
  URCEM_DEDICATED_STATION = 'URCEM_DEDICATED_STATION'
}

export enum CrossControlStatusType {
  NEW = 'NEW',
  FOLLOW_UP = 'FOLLOW_UP',
  CLOSED = 'CLOSED'
}

export enum CrossControlTargetType {
  VEHICLE = 'VEHICLE',
  COMPANY = 'COMPANY'
}

export enum CrossControlConclusionType {
  NO_FOLLOW_UP = 'NO_FOLLOW_UP',
  WITH_REPORT = 'WITH_REPORT'
}

export interface CrossControl {
  id?: string
  agentId?: number
  vesselId?: number
  serviceId?: number
  nbrOfHours?: number
  endDateTimeUtc?: string
  startDateTimeUtc?: string
  isSignedByInspector?: boolean
  type?: CrossControlTargetType
  origin?: CrossControlOriginType
  status?: CrossControlStatusType
  conclusion?: CrossControlConclusionType
  vesselName?: string
  isReferentialClosed?: boolean
  vesselExternalReferenceNumber?: string
}
