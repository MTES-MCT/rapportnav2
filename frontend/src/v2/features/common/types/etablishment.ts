export interface Establishment {
  id?: string
  name?: string
  siret?: string
  address?: string
}
