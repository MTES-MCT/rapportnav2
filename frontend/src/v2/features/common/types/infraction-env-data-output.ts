import { ControlType } from '@common/types/control-types'
import { FormalNoticeEnum, InfractionTypeEnum, VesselSizeEnum, VesselTypeEnum } from '@common/types/env-mission-types'

export interface InfractionEnvTargetOutput {
  vesselIdentifier: string
  companyName: string
  relevantCourt: string
  formalNotice: FormalNoticeEnum
  toProcess: boolean
  identityControlledPerson: string
  vesselType: VesselTypeEnum
  vesselSize: VesselSizeEnum
}

export interface InfractionEnvDataOutput {
  id?: string
  missionId?: number
  actionId?: string
  controlId?: string
  controlType?: ControlType
  infractionType?: InfractionTypeEnum
  observations?: string
  natinfs?: string[]
  target?: InfractionEnvTargetOutput
}
