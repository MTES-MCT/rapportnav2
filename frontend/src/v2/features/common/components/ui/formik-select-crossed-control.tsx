import { FormikSelect, FormikSelectProps } from '@mtes-mct/monitor-ui'
import styled from 'styled-components'
import { CrossControl } from '../../types/crossed-control-type'

export const FormikSelectCrossControl = styled(
  ({ crossControls, ...props }: Omit<FormikSelectProps, 'options'> & { crossControls: CrossControl[] }) => {
    return (
      <FormikSelect
        {...props}
        placeholder=""
        isLight={true}
        isRequired={true}
        isErrorMessageHidden={true}
        options={
          crossControls.map(v => ({
            value: v.id ?? '',
            label: `${v.vesselName} |Â ${v?.vesselExternalReferenceNumber} | Ouvert le ${v.startDateTimeUtc} |${v.status}`
          })) ?? []
        }
      />
    )
  }
)(() => ({}))
