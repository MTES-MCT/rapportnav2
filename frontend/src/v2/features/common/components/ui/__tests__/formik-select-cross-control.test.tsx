import { Formik } from 'formik'
import { vi } from 'vitest'
import { render } from '../../../../../../test-utils'
import { CrossControl } from '../../../types/crossed-control-type'
import { FormikSelectCrossControl } from '../formik-select-crossed-control'

const handleSubmit = vi.fn()
describe('FormikSearchVessel', () => {
  it('should match the snapshot', () => {
    const crossControls: CrossControl[] = [
      {
        id: '1dd960df-5c47-4eaf-ae25-37024589aed1',
        serviceId: 6
      }
    ]
    const wrapper = render(
      <Formik initialValues={{ vesselSize: '' }} onSubmit={handleSubmit}>
        <FormikSelectCrossControl
          name="crossControl.id"
          crossControls={crossControls ?? []}
          label="Rechercher la cible d’un contrôle en cours*"
        />
      </Formik>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
