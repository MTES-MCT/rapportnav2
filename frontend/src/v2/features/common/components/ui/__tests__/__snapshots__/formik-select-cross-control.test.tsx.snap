// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormikSearchVessel > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fhfEft ijxoir Element-Field Field-Select sc-Pgsbw"
      >
        <label
          class="sc-blIAwI ibbShc Element-Label"
          for="crossControl.id"
        >
          Rechercher la cible d’un contrôle en cours*
        </label>
        <div
          class="sc-lcItFd dsosWh"
        >
          <div
            class="rs-picker rs-picker-select rs-picker-default rs-picker-toggle-wrapper rs-picker-cleanable"
          >
            <div
              aria-controls="crossControl.id-listbox"
              aria-describedby="crossControl.id-describe"
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="listbox"
              class="rs-picker-toggle rs-btn rs-btn-default"
              id="crossControl.id"
              role="combobox"
              tabindex="0"
            >
              <div
                class="rs-stack"
                style="align-items: center;"
              >
                <div
                  class="rs-stack-item"
                  style="flex-grow: 1; overflow: hidden;"
                >
                  <input
                    aria-hidden="true"
                    class="rs-picker-toggle-textbox rs-picker-toggle-read-only"
                    data-testid="picker-toggle-input"
                    name="crossControl.id"
                    readonly=""
                    style="pointer-events: none;"
                    tabindex="-1"
                    value=""
                  />
                  <span
                    class="rs-picker-toggle-placeholder"
                    data-testid="picker-describe"
                    id="crossControl.id-describe"
                  >
                    Sélectionner
                  </span>
                </div>
                <div
                  class="rs-stack-item rs-picker-toggle-indicator"
                >
                  <svg
                    aria-hidden="true"
                    aria-label="angle down"
                    class="rs-picker-caret-icon rs-icon rs-icon"
                    data-category=""
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 21 32"
                    width="1em"
                  >
                    <path
                      d="M19.196 13.143a.612.612 0 01-.179.411l-8.321 8.321c-.107.107-.268.179-.411.179s-.304-.071-.411-.179l-8.321-8.321c-.107-.107-.179-.268-.179-.411s.071-.304.179-.411l.893-.893a.582.582 0 01.411-.179c.143 0 .304.071.411.179l7.018 7.018 7.018-7.018c.107-.107.268-.179.411-.179s.304.071.411.179l.893.893a.617.617 0 01.179.411z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fhfEft ijxoir Element-Field Field-Select sc-Pgsbw"
    >
      <label
        class="sc-blIAwI ibbShc Element-Label"
        for="crossControl.id"
      >
        Rechercher la cible d’un contrôle en cours*
      </label>
      <div
        class="sc-lcItFd dsosWh"
      >
        <div
          class="rs-picker rs-picker-select rs-picker-default rs-picker-toggle-wrapper rs-picker-cleanable"
        >
          <div
            aria-controls="crossControl.id-listbox"
            aria-describedby="crossControl.id-describe"
            aria-disabled="false"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="rs-picker-toggle rs-btn rs-btn-default"
            id="crossControl.id"
            role="combobox"
            tabindex="0"
          >
            <div
              class="rs-stack"
              style="align-items: center;"
            >
              <div
                class="rs-stack-item"
                style="flex-grow: 1; overflow: hidden;"
              >
                <input
                  aria-hidden="true"
                  class="rs-picker-toggle-textbox rs-picker-toggle-read-only"
                  data-testid="picker-toggle-input"
                  name="crossControl.id"
                  readonly=""
                  style="pointer-events: none;"
                  tabindex="-1"
                  value=""
                />
                <span
                  class="rs-picker-toggle-placeholder"
                  data-testid="picker-describe"
                  id="crossControl.id-describe"
                >
                  Sélectionner
                </span>
              </div>
              <div
                class="rs-stack-item rs-picker-toggle-indicator"
              >
                <svg
                  aria-hidden="true"
                  aria-label="angle down"
                  class="rs-picker-caret-icon rs-icon rs-icon"
                  data-category=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 21 32"
                  width="1em"
                >
                  <path
                    d="M19.196 13.143a.612.612 0 01-.179.411l-8.321 8.321c-.107.107-.268.179-.411.179s-.304-.071-.411-.179l-8.321-8.321c-.107-.107-.179-.268-.179-.411s.071-.304.179-.411l.893-.893a.582.582 0 01.411-.179c.143 0 .304.071.411.179l7.018 7.018 7.018-7.018c.107-.107.268-.179.411-.179s.304.071.411.179l.893.893a.617.617 0 01.179.411z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
